### PR TITLE
[#120] UI tests for deleting dataset, hiding data points, opening dataset and hiding dataset

### PR DESCRIPTION
**Pull request by Vatika1**
The UI testing was done through the selenium IDE. The test scripts are in Mocha JS so the file contains mocha packages.

### CODE REVIEW

**Review by Mehrdadmaskull — APPROVED** [review]
LGTM. But we need to add Mocha JS to our project later so that we'll be able to automate the UI tests too

File: Server/src/tests/unit/fileUploadController.test.js
Comment (GD-Dheer) [inline Server/src/tests/unit/fileUploadController.test.js]: These will likely need to get modified after the TS conversion. It wont be much but will require some code changes. @Mehrdadmaskull

File: Server/src/tests/unit/fileUploadController.test.js
Comment (GD-Dheer) [inline Server/src/tests/unit/fileUploadController.test.js]: Same as above

**Review by GD-Dheer — COMMENTED** [review]
Looks good. I would wait until TS is merged then update this PR and finally try to merge it. As it would require us to go back in the future anyway to update these tests

**Review by TiraFire — COMMENTED** [review]
No comment other than what GD said. Just fix the merge conflicts.

**Review by Mehrdadmaskull — DISMISSED** [review]
LGTM

**Review by TiraFire — APPROVED** [review]
Thank you for fixing the merge conflicts. Tests look thorough.

**Review by Mehrdadmaskull — APPROVED** [review]
Mocha is not part of our testing frameworks as of now so the UI tests gathered from Selenium WebDriver are not needed for now. We will consider including them once we have better test suites on both backend and frontend to automate the tests.

**Comment by GD-Dheer** [comment]
> Mocha is not part of our testing frameworks as of now so the UI tests gathered from Selenium WebDriver are not needed for now. We will consider including them once we have better test suites on both backend and frontend to automate the tests.

A-OK

**Review by GD-Dheer — APPROVED** [review]

**Pull request merged**
